### DISCUSSION TITLE
[Help]: Embla Carousel 2 row per slide item

### DISCUSSION THREAD

**Discussion by warrenvelasco** [davidjerleke/embla-carousel#816] [Help] [answered]
### Summary

I am searching ways  to do 2 item per slide in Embla Carousel but I can't find anything. 
I am working on a client that happens to use embla so I cant change the library for carousel

I want to achive same with this  reference from swiperjs https://swiperjs.com/demos/170-grid/core

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Additional information

![image](https://github.com/davidjerleke/embla-carousel/assets/64644452/181072b4-9e8e-4029-a402-36e4f427f1d5)

_No response_

**Comment by sarussss** [comment]
I think there are 2 ways to do so:
Method 1: You have to create html like that
Method 2: You create 2 sliders and then customize the js so they synchronize with each other

**Comment by warrenvelasco** [comment]
hi @sarussss 
Method one is out of option because I am getting the data from api and looping to `embla__slide`

I will try method 2. But do you happen to know how can I synchronize 2 seperate carousel?

**Reply by sarussss** [reply]
hi @warrenvelasco 
https://github.com/davidjerleke/embla-carousel/discussions/567

**Comment by davidjerleke** [comment]
@warrenvelasco you can achieve what you want with CSS. Just wrap every two slides in one slide and stack them vertically:

```html
<div class="embla">
  <div class="embla__container">
    <div class="embla__slide">
       <div class="embla__slide__item">Slide 1</div>
       <div class="embla__slide__item">Slide 2</div>
    </div>

    <div class="embla__slide">
       <div class="embla__slide__item">Slide 3</div>
       <div class="embla__slide__item">Slide 4</div>
    </div>

    …
  </div>
</div>
```

```css
.embla {
  overflow: hidden;
}
.embla__container {
  display: flex;
}
.embla__slide {
  flex: 0 0 calc(100% / 3);
  min-width: 0;
  display: flex;
  flex-direction: column;
}
```

And set your options to:

```tsx
const [emblaRef, emblaApi] = useEmblaCarousel({ align: 'start' });
```

**Reply by sarussss** [reply, marked as answer]
HI @davidjerleke 
Hi @warrenvelasco 
[Demo](https://stackblitz.com/edit/vitejs-vite-zmzrbf?file=index.html)

**Reply by sarussss** [reply]
That is method 1 that I mentioned, but it seems the author does not want to use this method.

**Reply by davidjerleke** [reply]
@sarussss thanks for all the help 🎉. @warrenvelasco I don’t understand what you mean by this:

> Method one is out of option because I am getting the data from api and looping to embla__slide

? Because even if slides are fetched dynamically from an API and created with a loop, method one is easy to achieve.

**Reply by warrenvelasco** [reply]
hi @davidjerleke @sarussss 
Thanks for suggestions. But I am getting my data from api and looping it to every slide. so technically I dont know how many slides item will be created.  Depends on the data from api. 

```
items = DATA FROM API

<div class="embla">
  <div class="embla__container">
    {items.map((item,index) => (
        <div key="{index}" class="embla__slide">
          <div class="embla__slide__item">{item}</div>
        </div>
    ))}
  </div>
</div>
```

**Reply by davidjerleke** [reply]
@warrenvelasco maybe I’m misunderstanding you, but by the time the loop renders you know what `items.length` is so you know how many groups of 2 you should render? And therefore it should be possible to wrap every 2 slides in a wrapper.

**Reply by warrenvelasco** [reply]
hi @davidjerleke @sarussss 
You are correct. your suggestions are the answer I need. I just need to find way to loop through  the data and yes I have access to`items.length`

Thank you so much guys. You are awesome!
